src/java.base/unix/native/libjava/TimeZone_md.c

```diff
@@ -555,58 +555,20 @@ findJavaTZ_md(const char *java_home_dir)
 /**
  * Returns a GMT-offset-based zone ID. (e.g., "GMT-08:00")
  */
-
-#if defined(MACOSX)
-
 char *
 getGMTOffsetID()
 {
-    time_t offset;
-    char sign, buf[32];
-    struct tm local_tm;
-    time_t clock;
-
-    clock = time(NULL);
-    if (localtime_r(&clock, &local_tm) == NULL) {
-        return strdup("GMT");
-    }
-    offset = (time_t)local_tm.tm_gmtoff;
-    if (offset == 0) {
+    char buf[32];
+    char offset[6];
+    struct tm localtm;
+    time_t clock = time(NULL);
+    if (localtime_r(&clock, &localtm) == NULL) {
         return strdup("GMT");
     }
-    if (offset > 0) {
-        sign = '+';
-    } else {
-        offset = -offset;
-        sign = '-';
-    }
-    sprintf(buf, (const char *)"GMT%c%02d:%02d",
-            sign, (int)(offset/3600), (int)((offset%3600)/60));
-    return strdup(buf);
-}
 
-#else
+    strftime(offset, 6, "%z", &localtm);
```

Comment:
Return value has to be examined. If it is not `5`, then I'd expect falling back to "GMT".

Reply:
Added check to verify output is as expected before proceeding, with fallback